### ISSUE TITLE
Which pitch detection algorithm does Vocaluxe use?

### ISSUE THREAD

**Issue by achimmihca** [Vocaluxe/Vocaluxe#520]
This is only a question:
What algorithm does Vocaluxe use for pitch detection? In the "PitchTracker" folder there is FFT stuff, dywapitchtrack, as well as some performous code for pitch detection.
Is Vocaluxe using a mix of these?

I ask because I am looking for a solid real-time pitch detection algorithm for the [UltraStar Play](https://github.com/UltraStar-Deluxe/Play) project. I tried autocorrelation, YIN (a little), some FFT stuff, and now dywapitchtrack that I just found here.
dywapitchtrack is giving really solid results so far, I am impressed.

Keep up the good work, Vocaluxe is a really cool project ;)

**Comment by Stefan1200-de** [comment]
Well, I'm quite new at Vocaluxe and I don't have any knowledge about pitch detection. And I guess that code was not touched since many years. Maybe @flokuep can say something about this.

**Comment by Jiiniasu** [comment]
I dug into this a couple of months ago, and your answer probably lies here:
https://github.com/Vocaluxe/Vocaluxe/blob/15d5b17fd2f8a0a2981ef2a24afcfb29ee0b26a7/Vocaluxe/Lib/Sound/Record/CBuffer.cs#L27

Description on what it is in comment at
https://github.com/Vocaluxe/Vocaluxe/blob/15d5b17fd2f8a0a2981ef2a24afcfb29ee0b26a7/PitchTracker/ptAKF.h#L23

**Comment by achimmihca** [comment]
Ok thx.
It looks like CBuffer.cs tests different algorithms. It seems that
- CAnalyzer is the performous implementation (looks like its doing FFT stuff)
- AKF I guess is autocorrelation function / amplitude magnitude difference function (maybe similar to the original USDX algorithm)
- DyWa is the dywapitchtrack implementation (its core is a wavelet algorithm)
- CPtSharp does not seem to be in this repository (I did not find anything with that name)?

Maybe Vocaluxe uses a mixture of these.

Anyway, for me the dywapitchtrack algorithm is the most interesting because I did not come across such a wavelet algorithm before. Furthermore, the implementation has a fitting license (MIT) for our project and from my few experiments it seems to be very fast and pretty accurate.

FFT stuff I tried before, just to find out that FFT has pretty bad resolution in the low frequency range that is human voice. Autocorrelation is used at the moment for USPlay but it might be that dywapitchtrack can compete very well.

Again, thanks for the help and keep up the good work.
